sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -160,6 +160,31 @@ func newLanguageHost(nodePath, runPath, engineAddress,
 	}
 }
 
+func compatibleVersions(a, b semver.Version) (bool, string) {
+	switch {
+	case a.Major == 0 && b.Major == 0:
+		// If both major versions are pre-1.0, we require that the major and minor versions match.
+		if a.Minor != b.Minor {
+			return false, "Differing major or minor versions are not supported."
+		}
+
+	case a.Major >= 1 && b.Major >= 1:
+		// If both major versions are post-1.0, we require that the major versions match.
+		if a.Major != b.Minor {
+			return false, "Differing major versions are not supported."
+		}
+
+	case a.Major == 1 && b.Minor == 17 || b.Major == 1 && a.Minor == 17:
```

Comment:
Might want to confirm that when you check the minor for 17, the major is 0 here, right?

Reply:
No need, but I've added a check for clarity.